app/src/main/java/net/squanchy/typeface/TypefaceController.kt

```diff
@@ -0,0 +1,26 @@
+package net.squanchy.typeface
+
+import android.graphics.Typeface
+import android.text.Spannable
+import android.text.SpannableString
+import android.text.Spanned
+
+class TypefaceController {
+
+    fun applyTypeface(text: CharSequence, typeface: Typeface) =
+            if (text !is Spannable) {
+                addSpanTo(SpannableString(text), typeface)
+            } else {
+                addSpanTo(text, typeface)
+            }
+
+
+    private fun addSpanTo(text: Spannable, typeface: Typeface) =
+            text.apply {
+                text.setSpan(CustomTypefaceSpan(typeface), 0, text.length, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
```

Comment:
You don't need to repeat `text` if you use `apply`, `text` is the receiver. That's the main reason to use `apply` in the first place :)